accepted/0000-buildpack-set-env-vars-defaults.md

```diff
@@ -0,0 +1,153 @@
+# Default Values for Language Ecosystem Env Vars Set by Buildpacks
+
+## Summary
+
+{{A concise, one-paragraph description of the change.}}
+
+## Motivation
+
+{{Why are we doing this? What pain points does this resolve? What use cases
+does it support? What is the expected outcome? Use real, concrete examples to
+make your case!}}
+a mostly-prose explanation of why we should do this.
+
+## Detailed Explanation
+
+Some Paketo buildpacks set environment variables used to configure language
+ecosystem tooling. Often, buildpacks use the
+[override](https://github.com/buildpacks/spec/blob/main/buildpack.md#override)
+environment variable modification rule provided by the lifecycle. As a result,
+buildpack users **cannot** themselves provide alternate values for these
+environment variables at build or launch time (e.g. with `pack build app --env
+VARIABLE=value`).
+
+
+If buildpacks instead set language-ecosystem environment variables using the
+[default](https://github.com/buildpacks/spec/blob/main/buildpack.md#default)
+environment variable modification rule, buildpack users will consistently be
+able to override these environment variables at build and/or launch to better
+meet their needs.
+
+Additionally, the each buildpack should check whether there is already a value
+for the given environment variable available, and if there is, it should **set
+that value as the default** for the remainder of the build and/or for launch.
```

Comment:
> If the variable is set during why would the buildpack need to set it as the default for the remainder of bulid.
Good point. More accurately, the proposed change would have the buildpack set that value as the default for launch.

That notwithstanding +1 to the idea that a buildpack user might want (maybe usually wants?) build- and launch-time values of an environment variable to differ.

Reply:
+1 to Emily's second point. I agree that it could be confusing to set a value at build-time and have that carry over to launch-time. In the case where a user would want the same variable at build- and launch-time they'd have to set it twice. Possibly redundant, yes, but it makes for a clearer experience while also covering the other cases.